crates/legacy/src/service.rs

```diff
@@ -1202,209 +1021,68 @@ pub async fn run_non_permissioned_standalone_builder_service<
     // sending a Decide event from the hotshot to the builder states
     decide_sender: BroadcastSender<MessageType<Types>>,
 
-    // Url to (re)connect to for the events stream
-    hotshot_events_api_url: Url,
+    // HotShot event stream
+    hotshot_event_stream: impl Stream<Item = Event<Types>>,
```

Comment:
We could use a generic placeholder for this `Stream` so that we don't need to reference it via an `impl`.  It does require adding another generic parameter to the function however.

```suggestion
pub async fn run_non_permissioned_standalone_builder_service<
    Types: NodeType,
    Ver: StaticVersionType,
    S: Stream<Item = Event<Types>> + Unpin
>(
    // sending a DA proposal from the hotshot to the builder states
    da_sender: BroadcastSender<MessageType<Types>>,

    // sending a Quorum proposal from the hotshot to the builder states
    quorum_sender: BroadcastSender<MessageType<Types>>,

    // sending a Decide event from the hotshot to the builder states
    decide_sender: BroadcastSender<MessageType<Types>>,

    // HotShot event stream
    hotshot_event_stream: S,
    ...
```

This would allow the Stream's size to be known at compile time.

Reply:
Additionally, I'm not certain about this type signature.  Since we're consuming from the Events service, the `Item`s contained within the `Stream` all tend to be `Result`s instead of just `Event`s.  This would require the caller to somehow resolve these `Event`s from the `Result`s making the caller have to do extra work.